### ISSUE TITLE
Adding prev history to locationState on initial load?

### ISSUE THREAD

**Issue by ml242** [faceyspacey/redux-first-router#361]
I'd like to push some history (e.g. pathname) to the prev object on initial load of the page.

Is it possible to do this without using the Rudy-History package or do I need the createHistory function that returns an object in some proper form to connectRoutes?

Thanks for shedding a little light on this, it seems weird to require another package when what it looks like I'm trying to do is simply:

```
location: {
  ...location,
  prev: {
    ...location.prev,
    pathname: 'newpath'
  }
}
```

**Comment by hedgepigdaniel** [comment]
I guess you could create your own location reducer that wraps the redux-first-router one.

But why do you want to do this?

**Comment by ml242** [comment]
if the app opens in a new window and you want to be able to be able to give the user the ability to nav back to a central location like the store. I know this isn't typical behavior but this is what I'm looking to do.

**Comment by hedgepigdaniel** [comment]
I haven't tried it, but my impression from reading the code is that this is supported in Rudy by using the history `reset` function

https://github.com/respond-framework/rudy/blob/3c184800682bbfa5adaecb7e8aef3657cb0ff9f6/packages/rudy/src/middleware/transformAction/utils/formatAction.js#L30

https://github.com/respond-framework/rudy/blob/3c184800682bbfa5adaecb7e8aef3657cb0ff9f6/packages/rudy/src/actions/history.js#L34

Basically you import `reset` and it will create a redux action that resets the entire history stack to whatever you want

**Comment by ml242** [comment]
Thanks for pointing that out, I'm going to give it a try.

**Comment by ml242** [comment]
It seems like no matter what i try with regard to pushing/replacing/resetting state on initial page load, i dont get the back button functionality (it's grayed out).

But if I browse to a page and then back, the new history has been inserted and takes the correct number of button presses (so one more will go to the new Page / button is active).

Do you think this is a chrome issue and not a router issue?

Thanks again
